### PR TITLE
add ODE benchmarks

### PR DESCRIPTION
**Pull request by Dpananos**
Solves  #3694

Should draw 3000 samples (1000 warm up in two chains) in approx 2.5 minutes, maybe a little quicker.  
I've tried to follow the example of `ExampleSuite`.  Let me know if I've missed something or if you prefer something else.

### CODE REVIEW

File: benchmarks/benchmarks/benchmarks.py [hunk 1]
```diff
@@ -195,3 +195,58 @@ def track_glm_hierarchical_ess(self, step):
 
 
 CompareMetropolisNUTSSuite.track_glm_hierarchical_ess.unit = 'Effective samples per second'
+
+
+class DifferentialEquationSuite:
+    """Implements ode examples to keep up with benchmarking them."""
+
+    timeout = 360
```
Comment (michaelosthege) [inline benchmarks/benchmarks/benchmarks.py]: The timeout could be too tight

**Review by michaelosthege — CHANGES_REQUESTED** [review]

File: benchmarks/benchmarks/benchmarks.py (hunk 1 above)
Comment (Dpananos) [reply benchmarks/benchmarks/benchmarks.py]: Resolved by latest commit.

**Comment by michaelosthege** [comment]
LGTM

**Review by ColCarroll — APPROVED** [review]

**Comment by ColCarroll** [comment]
Thanks, @Dpananos -- the cloud benchmark machine is currently down, but I've been running on my laptop every week or so.

**Pull request merged**